include/highfive/H5DataSet.hpp

```diff
@@ -90,8 +90,8 @@ class DataSet : public Object,
         return getSpace().getElementCount();
     }
 
-    // No empty datasets
-    DataSet() = delete;
+    H5_DEPRECATED("Default ctor creates unsafe uninitialized objects")
```

Comment:
```suggestion
    H5_DEPRECATED("Default constructor creates unsafe uninitialized objects")
```

Reply:
Yep, relatively to the latest release we had removed API without deprecating, which is not nice (my bad). We only noticed in some internal projects. The added tests would fail without the patch.